AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncModelMetadata.swift

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+
+/// Metadata that contains information about an associated parent object.
+public struct AppSyncModelMetadata: Codable {
+    let appSyncAssociatedId: String
+    let appSyncAssociatedField: String
+    let apiName: String?
+
+    init(associatedId: String, associatedField: String, apiName: String?) {
+        self.appSyncAssociatedId = associatedId
+        self.appSyncAssociatedField = associatedField
+        self.apiName = apiName
+    }
```

Comment:
nit: if you're not changing the access level from `internal`, you don't need to explicitly write it: the compiler auto-synthesizes a member-wise initializer for structs.